calcom/cal.com
# fix: org username check [CAL-2352]

## What does this PR do?

When a user was invited to an org, the signup page was not reflecting the organization context in which the new user was in. We now rely on the assigned team in the invitation token and in the refactored logic to check if a username exists in the organization when being created. Also, org username when signing up using a token and then in profile page is not modifiable.

## Type of change


## How should this be tested?

Please test the 3 scenarios involved in the introduced changes:
 - Create a new regular user using web app signup page
 - Invite someone to a regular team and have them follow the invite link, they should be able to create a new user and username is checked accordingly with no scope and is modifiable
 - Invite someone to an org and have them follow the invite link, they should be able to create a new user and username is check accordingly scoped to org and not modifiable

## Mandatory Tasks
